### PR TITLE
Notebook integration test

### PR DESCRIPTION
**Pull request by froesef**
The following adds an integration test that adds the data.csv file to an S3 instance in the kind cluster, deploys an application that is using an arrow flight service to read data and reads the data using this arrow-flight service to validate that it's correct.

The goal of this test is to do and end to end test with the original components rather than mocked components. In future we could also integrate a copy from one bucket to the next to also use the implicit copy module.

### CODE REVIEW

File: manager/controllers/app/fybrik_notebook_test.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	RegisterFailHandler(Fail)
+	defer GinkgoRecover()
+
+	// Copy data.csv file to S3
+	// S3 is assumed to be exposed on localhost at port 9090
+	region := "us-east-1"
+	endpoint := "http://localhost:9090"
+	bucket := "bucket1"
+	key1 := "data.csv"
+	filename := "/Users/ffr/go/src/github.com/fybrik/fybrik/samples/kubeflow/data.csv"
```
Comment (revit13) [inline manager/controllers/app/fybrik_notebook_test.go:53]: I guess this should not be hard coded...

File: manager/controllers/app/fybrik_notebook_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+func TestS3Notebook(t *testing.T) {
+	if s, ok := os.LookupEnv("VALUES_FILE"); !ok || s != "charts/fybrik/notebook-tests.values.yaml" {
+		t.Skip("Only executed for notebook tests")
+	}
+	RegisterFailHandler(Fail)
+	defer GinkgoRecover()
+
+	// Copy data.csv file to S3
+	// S3 is assumed to be exposed on localhost at port 9090
+	region := "us-east-1"
```
Comment (revit13) [inline manager/controllers/app/fybrik_notebook_test.go:49]: should it be identical to the one in the `Asset` resource (theshire)?

File: manager/testdata/notebook/setup.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+kubectl label deployment my-notebook app.kubernetes.io/name=my-notebook
+kubectl wait --for=condition=available --timeout=120s deployment/my-notebook
+kubectl expose deployment my-notebook --port=80 --target-port=8888
+
+# Create asset and secret
+kubectl -n fybrik-notebook-sample apply -f example-asset.yaml
+kubectl -n fybrik-notebook-sample apply -f s3credentials.yaml
+
+# TODO Use remote version when compatible
+# kubectl apply -f https://github.com/fybrik/arrow-flight-module/releases/latest/download/module.yaml -n fybrik-system
```
Comment (revit13) [inline manager/testdata/notebook/setup.sh:20]: I guess it should be `https://raw.githubusercontent.com/fybrik/arrow-flight-module/master/module.yaml`

File: manager/testdata/notebook/setup.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+kubectl apply -n fybrik-system -f arrow-flight-module.yaml
+
+kubectl -n fybrik-system create configmap sample-policy --from-file=sample-policy.rego
+kubectl -n fybrik-system label configmap sample-policy openpolicyagent.org/policy=rego
+while [[ $(kubectl get cm sample-policy -n fybrik-system -o 'jsonpath={.metadata.annotations.openpolicyagent\.org/policy-status}') != '{"status":"ok"}' ]]; do echo "waiting for policy to be applied" && sleep 5; done
```
Comment (revit13) [inline manager/testdata/notebook/setup.sh:25]: btw, it takes quite a while for the status to change, is there a way to preload the policies somehow?

File: manager/controllers/app/fybrik_notebook_test.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+		if err != nil {
+			return false
+		}
+		return application.Status.Ready
+	}, timeout, interval).Should(Equal(true))
+
+	// Forward port of arrow flight service to local port
+	svcName := strings.Replace(application.Status.AssetStates["fybrik-notebook-sample/data-csv"].Endpoint.Hostname, ".fybrik-blueprints.svc.cluster.local", "", 1)
+
+	command := exec.Command("kubectl", "-n", BlueprintNamespace, "port-forward", "svc/"+svcName, "8083:80")
```
Comment (revit13) [inline manager/controllers/app/fybrik_notebook_test.go:155]: the port also should be taken from the endpoint?

**Review by revit13 — APPROVED** [review]

File: manager/controllers/app/fybrik_notebook_test.go (hunk 5 above)
Comment (shlomitk1) [reply manager/controllers/app/fybrik_notebook_test.go:155]: No, it is port-forward to the port above

**Comment by froesef** [comment]
Merging this now in order to continue the work in refactoring branch https://github.com/fybrik/fybrik/pull/777

**Pull request merged**